File: leetcode_179. Largest Number_3lines.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
       sort(nums.begin(), nums.end(), [](int a, int b){ return to_string(a) + to_string(b) > to_string(b) + to_string(a);});
       string ans = accumulate(nums.begin(), nums.end(), string(), [](string &r, int a){return r + to_string(a);});
       return ans[0] == '0' ? "0" : ans;
       }
};

File: leetcode_179. Largest Number_3lines.py
class Solution:
    def largestNumber(self,nums):
        num_to_str = ''.join(sorted([str(x) for x in nums],cmp=lambda a,b:cmp(b+a,a+b)))
        return num_to_str if num_to_str[0] !='0' else "0"